src/Main/Main.tsx

```diff
@@ -1,141 +1,177 @@
+  context,
+  drawerContent,
+  children,
+  headerLeftContent,
+  headerRightContent,
+  open: openOverride = false,
+}: Props): JSX.Element => {
+  const [open, setOpen] = React.useState(true);
+
+  const handleDrawerClose = (): void => {
+    setOpen(false);
   };
 
-  handleDrawerClose = (): void => {
-    this.setState({ open: false });
-    this.props.handleDrawerClose?.();
+  const handleDrawerToggle = (): void => {
+    setOpen((state) => !state);
   };
 
-  render(): JSX.Element {
-    const {
-      children,
-      fullScreen,
-      headerLeftContent,
-      headerRightContent,
-      headerCenterContent,
-      headerId,
-      menuButtonId,
-      sidebar,
-    } = this.props;
-    const { open } = this.state;
-    const hasSidebar = Boolean(sidebar);
-
-    return (
-      <StyledRoot>
-        <Header
-          context={this.props.context}
-          hasSidebar={hasSidebar}
-          isSidebarOpen={open}
-          handleDrawerOpen={this.handleDrawerOpen}
-          handleDrawerClose={this.handleDrawerClose}
-          leftContent={headerLeftContent}
-          rightContent={headerRightContent}
-          centerContent={headerCenterContent}
-          id={headerId}
-          menuButtonId={menuButtonId}
-          sx={this.props.headerSx}
-        />
-
-        {hasSidebar && <Sidebar isSidebarOpen={open}>{sidebar}</Sidebar>}
-
-        <StyledMain open={open} fullScreen={fullScreen}>
-          <>
-            <DrawerHeaderContainer />
-            {children}
-          </>
-        </StyledMain>
-      </StyledRoot>
-    );
-  }
-}
+  React.useEffect(() => {
+    setOpen(openOverride);
+  }, [openOverride]);
+
+  return (
+    <Box sx={{ display: 'flex' }} height='100vh' position='absolute'>
+      <CssBaseline />
+      <AppBar
+        position='fixed'
+        sx={{
+          background: context
+            ? buildHeaderGradient(AccentColors[context])
+            : PRIMARY_COLOR,
+        }}
+      >
+        <Toolbar>
+          <Stack
+            width='100%'
+            direction='row'
+            alignItems='center'
+            justifyContent='space-between'
+            spacing={2}
+            flex={1}
+            minWidth='0px'
+          >
+            <Stack direction='row' alignItems='center' flex={1} minWidth='0px'>
+              <IconButton
+                color='inherit'
+                aria-label='open drawer'
+                onClick={handleDrawerToggle}
+                edge='start'
+                sx={{ mr: 2 }}
+              >
+                <MenuIcon />
+              </IconButton>
+              {headerLeftContent}
+            </Stack>
+            {headerRightContent}
+          </Stack>
+        </Toolbar>
+      </AppBar>
+      <Box component='nav' aria-label='navigation'>
+        <Drawer
+          variant='temporary'
+          open={open}
+          onClose={handleDrawerClose}
+          ModalProps={{
+            keepMounted: true, // Better open performance on mobile.
```

Comment:
Does it make sense to have it false on desktop then? Just asking.

Reply:
Actually this just means that we use the `keepMounted` option of the dialog. So it will not be destroyed when closed. This is to improve performance on mobile. It also improves performance on desktop. But in this case this specific variant of the drawer is not shown on large screens (display: none).